netbox/extras/forms/bulk_import.py

```diff
@@ -160,14 +161,55 @@ class Meta:
 
 
 class ConfigTemplateImportForm(CSVModelForm):
+    data_source = CSVModelChoiceField(
+        label=_('DataSource'),
+        queryset=DataSource.objects.all(),
+        required=False,
+        to_field_name='name',
+        help_text=_('data_source of the config template')
+    )
+
+    data_file = CSVModelChoiceField(
+        label=_('DataFile'),
+        queryset=DataFile.objects.all(),
+        required=False,
+        to_field_name='path',
+        help_text=_('DataFile containing the template code')
+    )
+
+    template_code = forms.CharField(
+        label=_('Template code'),
+        required=False,
+        widget=forms.Textarea(attrs={'class': 'font-monospace'})
+    )
+
+    auto_sync_enabled = forms.BooleanField(
+        required=False,
+        label=_('auto sync enabled'),
+        help_text=_("Enable automatic synchronization of data when the data file is updated")
+    )
 
     class Meta:
         model = ConfigTemplate
         fields = (
-            'name', 'description', 'template_code', 'environment_params', 'mime_type', 'file_name', 'file_extension',
-            'as_attachment', 'tags',
+            'name', 'description', 'template_code', 'data_source', 'data_file',
+            'auto_sync_enabled', 'environment_params', 'mime_type', 'file_name',
+            'file_extension', 'as_attachment', 'tags',
         )
 
+    def clean_template_code(self):
+        # Make sure template_code is None when it's not included in the uploaded data
+        if not self.data.get('template_code') and not self.data.get('data_file'):
+            raise forms.ValidationError(_("Must specify either local content or a data file"))
+
+        return self.cleaned_data['template_code']
```

Comment:
This logic should be run under the form's `clean()` method as it involves more than one field.